src/infra/filesQueryRepo.go

```diff
@@ -0,0 +1,197 @@
+package infra
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/speedianet/os/src/domain/entity"
+	"github.com/speedianet/os/src/domain/valueObject"
+	infraHelper "github.com/speedianet/os/src/infra/helper"
+)
+
+type FilesQueryRepo struct{}
+
+func (repo FilesQueryRepo) unixFileFactory(
+	filePath valueObject.UnixFilePath,
+) (entity.UnixFile, error) {
+	var unixFile entity.UnixFile
+
+	fileInfo, err := os.Stat(filePath.String())
+	if err != nil {
+		return unixFile, err
+	}
+
+	fileSysInfo := fileInfo.Sys().(*syscall.Stat_t)
+
+	unixFileUid, err := valueObject.NewUnixUid(fileSysInfo.Uid)
+	if err != nil {
+		return unixFile, err
+	}
+
+	fileOwner, err := user.LookupId(unixFileUid.String())
+	if err != nil {
+		return unixFile, err
+	}
+
+	unixFileUsername, err := valueObject.NewUsername(fileOwner.Username)
+	if err != nil {
+		return unixFile, err
+	}
+
+	unixFileGid, err := valueObject.NewGroupId(fileSysInfo.Gid)
+	if err != nil {
+		return unixFile, err
+	}
+
+	fileGroupName, err := user.LookupGroupId(unixFileGid.String())
+	if err != nil {
+		return unixFile, err
+	}
+
+	unixFileGroup, err := valueObject.NewGroupName(fileGroupName.Name)
+	if err != nil {
+		return unixFile, err
+	}
+
+	unixFileAbsPath, err := filepath.Abs(filePath.String())
+	if err != nil {
+		return unixFile, err
+	}
+
+	unixFilePath, err := valueObject.NewUnixFilePath(unixFileAbsPath)
+	if err != nil {
+		return unixFile, err
+	}
+
+	var unixFileExtensionPtr *valueObject.UnixFileExtension
+	unixFileExtension, err := unixFilePath.GetFileExtension()
+	if err != nil {
+		unixFileExtensionPtr = nil
+	}
+
+	unixFileExtensionPtr = &unixFileExtension
+	if unixFileExtension.String() == "" {
+		unixFileExtensionPtr = nil
+	}
+
+	filePermissions := fileInfo.Mode().Perm()
+	filePermissionsStr := fmt.Sprintf("%o", filePermissions)
+	unixFilePermissions, err := valueObject.NewUnixFilePermissions(filePermissionsStr)
+	if err != nil {
+		log.Print(err)
```

Comment:
Eu havia mencionado na review anterior para só usar log em factory caso seja um for loop, o que não é o caso aqui.

Reply:
Falta de atenção minha. Ajustado!